skbio/tree/_tree.py

```diff
@@ -2327,7 +2327,11 @@ def _set_max_distance(self):
                 if len(n.children) == 1:
                     raise TreeError("No support for single descedent nodes")
                 else:
-                    tip_info = [(max(c.MaxDistTips), c) for c in n.children]
+                    tip_info = []
+                    for c in n.children:
+                        (left_d, _), (right_d, _) = c.MaxDistTips
```

Comment:
:+1: , the existing list comprehension can just be modified to pass `key`, seems the simplest and most Pythonic IMO